src/robot/htmldata/rebot/model.js

```diff
@@ -272,36 +280,55 @@ window.stats = (function () {
             });
     }
 
-    function calculatePercents(total, passed, failed) {
+    function calculatePercents(total, passed, failed, skipped) {
         if (total == 0)
-            return [0.0, 0.0];
+            return [0.0, 0.0, 0.0];
         var pass = 100.0 * passed / total;
         var fail = 100.0 * failed / total;
+        var skip = 100.0 * skipped / total;
+        
         if (pass > 0 && pass < 0.1)
-            return [0.1, 99.9];
+            if (fail > 0 && fail < 0.1)
+                return [0.1, 0.1, 99.8];
+            if (skip > 0 && skip < 0.1)
+                return [0.1, 99.9, 0.1];
+
         if (fail > 0 && fail < 0.1)
-            return [99.9, 0.1];
-        return [Math.round(pass*10)/10, Math.round(fail*10)/10];
+            if (pass > 0 && pass < 0.1)
+                return [0.1, 0.1, 99.8];
+            if (skip > 0 && skip < 0.1)
+                return [99.8, 0.1, 0.1];
+
+        if (skip > 0 && skip < 0.1)
+            if (pass > 0 && pass < 0.1)
+                return [0.1, 99.8, 0.1];
+            if (fail > 0 && fail < 0.1)
```

Comment:
Note that this if might fall outside the block as the indentation suggests. Please use `{` and `}`.